### PR TITLE
feat: replace multiple attributions at once

### LINKED ISSUE #2467: Add option to replace multiple attributions at the same time

### PR DESCRIPTION
**Pull request by mstykow**
### Summary of changes

- add button to replace selected attributions in attribution list
- refactor popup to replace attributions to be able to handle multiple
- allow virtualized list to have a min-height
- make click action on package card optional so that the pointer cursor and hover state also become optional

image

### Context and reason for change

closes #2467

### How can the changes be tested

Replace attributions in attribution view. Some edge cases to test:
- It's now possible to replace an attribution with a pre-selected one (previously impossible). In this case, the pre-selected attribution should become confirmed automatically when the replacement process finishes.
- When you replace an attribution, you should see the target attribution details in the attribution column.
- After you replace an attribution, none of the select checkboxes in the attribution list should be active.
- The replace popup button should be disabled when there is nothing selected or when there are no unselected attributions left.

### CODE REVIEW

**Review by leslielazzarino — APPROVED** [review]

File: src/Frontend/Components/ReplaceAttributionsPopup/ReplaceAttributionsPopup.tsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    () =>
+      getFilteredAndSortedPackageCardIdsAndDisplayPackageInfos(
+        filteredAttributions,
+        search,
+        false,
+      ),
+    [filteredAttributions, search],
+  );
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (mstykow) [reply src/Frontend/Components/ReplaceAttributionsPopup/ReplaceAttributionsPopup.tsx:67]: `useLayoutEffect` is an extremely uncommon hook and according to the React docs best avoided. the use cases for it is to avoid flashing effects when UI elements need to be moved around the screen due to a re-measurment occuring after the first paint. so no, cleaning up internal state is not the right situation to use this kind of hook in.

[the remark this answers is not in the document]
Comment (mstykow) [reply src/Frontend/Components/ReplaceAttributionsPopup/ReplaceAttributionsPopup.tsx:67]: https://react.dev/reference/react/useLayoutEffect#usage

[the remark this answers is not in the document]
Comment (leslielazzarino) [reply src/Frontend/Components/ReplaceAttributionsPopup/ReplaceAttributionsPopup.tsx:67]: Interesting, still in this kind of situation I believe it should improve performances. I will test my assumption when I have time, for now let's keep using useEffect.
